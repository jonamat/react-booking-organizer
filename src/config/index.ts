import { WeekDays, HourMinute } from '../types';
let counter = 0;

/* -------------------------------------------------------------------------- */
/*                           Database configuration                           */
/* -------------------------------------------------------------------------- */

// IDs of the databases collections
export const RESERVATIONS_COLLECTION = 'reservations';
export const EMPLOYEES_COLLECTION = 'employees';
export const SERVICES_COLLECTION = 'services';
export const CUSTOMERS_COLLECTION = 'customers';

// Credentials for the sandbox database
export const SANDBOX_USERNAME = 'sandbox@demo.db';
export const SANDBOX_PASSWORD = 'sandbox';

/* -------------------------------------------------------------------------- */
/*                              App configuration                             */
/* -------------------------------------------------------------------------- */

// Key name of the employees order array in local storage
export const LS_EMPLOYEES_ORDER_KEY_NAME = 'employees_order';

// Minimum subdivision in minutes for the agenda. It should be a divisor of 60
export const MIN_STEP = 15;

// Hours of the longest service offered
export const MAX_SERVICE_DURATION_MINS = 3 * 60;

// Opening hour in HourMinute { hours(24), minutes }
export const OPENING_TIME: HourMinute = { hour: 8, minute: 0 };

// Closing hour in HourMinute { hours(24), minutes }
export const CLOSING_TIME: HourMinute = { hour: 20, minute: 0 };

/* -------------------------------------------------------------------------- */
/*                          Theming and localization                          */
/* -------------------------------------------------------------------------- */

// Progression of days according to the system
export const SYSTEM_ORDERED_DAYS: Array<WeekDays> = [
    'sunday',
    'monday',
    'tuesday',
    'wednesday',
    'thursday',
    'friday',
    'saturday',
];

// Progression of days according to the user local regulation (it_IT)
export const WEEK_DAYS: Array<WeekDays> = [
    'monday',
    'tuesday',
    'wednesday',
    'thursday',
    'friday',
    'saturday',
    'sunday',
];

// Reservation table sizes in px
export const CELLS_HEIGHT = 35;
export const CELLS_WIDTH = 200;
export const HEAD_COL_WIDTH = 70;

// Translations for material table
export const MATERIAL_TABLE_LOCALIZATION = {
    pagination: {
        labelRowsSelect: 'righe',
        labelRowsPerPage: 'Righe per pagina',
        firstAriaLabel: 'Prima pagina',
        firstTooltip: 'Prima pagina',
        previousAriaLabel: 'Pagina precedente',
        previousTooltip: 'Pagina precedente',
        nextAriaLabel: 'Pagina successiva',
        nextTooltip: 'Pagina successiva',
        lastAriaLabel: 'Ultima pagina',
        lastTooltip: 'Ultima pagina',
        labelDisplayedRows: '{from}-{to} di {count}',
    },
    toolbar: {
        searchTooltip: 'Cerca',
        searchPlaceholder: 'Cerca',
    },
    header: {
        actions: 'Azioni',
    },
    body: {
        emptyDataSourceMessage: 'Nessun elemento trovato',
        filterRow: {
            filterTooltip: 'Filtra',
        },
    },
};

// App date format
export const TIME_DATE_FORMAT = {
    weekday: 'long',
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: 'numeric',
    minute: 'numeric',
};

/* -------------------------------------------------------------------------- */
/*                    Constants derived from previous vars                    */
/* -------------------------------------------------------------------------- */

// Steps which has 1 hour
export const SEGMENTS_IN_1HOUR = 60 / MIN_STEP;
counter = 0;

// Array of progressive minutes steps within a hour
export const PROGRESSIVE_MINS_WITHIN_HOUR = new Array<number>(SEGMENTS_IN_1HOUR)
    .fill(0)
    .map(() => MIN_STEP * counter++);
counter = 0;

// Array of progressive minutes steps within MAX_SERVICE_DURATION_MINS without the first element (zero)
export const PROGRESSIVE_MINS_WITHIN_MAX_SERVICE = new Array<number>(
    Math.floor(MAX_SERVICE_DURATION_MINS / MIN_STEP) + 1,
)
    .fill(0)
    .map(() => MIN_STEP * counter++)
    .slice(1);
counter = 0;

// Array of progressive HourMinute steps within a business day
export const PROGRESSIVE_HOURS_WITHIN_BUSINESS_DAY: Array<HourMinute> = new Array<string>(
    Math.floor(
        (Math.abs(CLOSING_TIME.hour - OPENING_TIME.hour) * 60 +
            Math.abs(CLOSING_TIME.minute - OPENING_TIME.minute) +
            MIN_STEP) /
            MIN_STEP,
    ),
)
    .fill('')
    .map(() => {
        const totalMinutes = MIN_STEP * counter++;
        const hour = Math.floor(totalMinutes / 60) + OPENING_TIME.hour;
        const minute = (totalMinutes % 60) + OPENING_TIME.minute;
        return { hour, minute };
    });
