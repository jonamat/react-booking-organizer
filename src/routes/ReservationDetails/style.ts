import { makeStyles, Theme } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) => ({
    container: {
        padding: theme.spacing(2),
    },
    header: {
        display: 'flex',
        paddingTop: theme.spacing(2),
        paddingBottom: theme.spacing(2),
    },
    subtitle: {
        fontWeight: theme.typography.fontWeightMedium,
    },
    grow: {
        flexGrow: 1,
    },
}));

export default useStyles;
