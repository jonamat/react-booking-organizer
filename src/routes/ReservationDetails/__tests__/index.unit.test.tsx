/**
 * Unit test suite
 *
 * Target: ReservationDetails component
 * Tests:
 * - Data position correctness
 * - Redirection
 * - Snapshot matching
 * - Functionality
 */

// Test resources
import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider, createGenerateClassName } from '@material-ui/core';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { addMinutes } from 'date-fns';

// Test configuration
import {
    __mockCustomers,
    __mockServices,
    __mockEmployees,
    __mockReservations,
    __setUseStore,
} from '../../../__mocks__/react-redux';
import { TIME_DATE_FORMAT } from '../../../config';

// Mocked modules
import { toast } from 'react-toastify';

// Component to test
import ReservationDetails from '..';

const match = {
    params: {
        reservationId: __mockReservations[0].id,
    },
};

/* -------------------------------------------------------------------------- */
/*                                    Mocks                                   */
/* -------------------------------------------------------------------------- */

jest.mock('../../../utils', () => ({
    ...(jest.requireActual('../../../utils') as Record<string, any>),
    deleteReservation: jest.fn(),
}));
const spyTError = jest.spyOn(toast, 'error');
const mockOpenDialog = jest.fn();
jest.mock('redux', () => ({
    ...(jest.requireActual('redux') as Record<string, any>),
    bindActionCreators: () => ({
        openDialog: mockOpenDialog,
        closeDialog: () => undefined,
    }),
}));
const history = {
    push: jest.fn(),
    goBack: jest.fn(),
};

/* -------------------------------------------------------------------------- */
/*                              Bridge operations                             */
/* -------------------------------------------------------------------------- */

beforeEach(() => {
    jest.clearAllMocks();
});

/* -------------------------------------------------------------------------- */
/*                               Render wrapper                               */
/* -------------------------------------------------------------------------- */

/**
 * Render the component with the custom configuration
 * @param props Component props
 */
const renderComponent = (props?) => {
    return render(<ReservationDetails match={match} history={history} {...props} />);
};

/* -------------------------------------------------------------------------- */
/*                                    Tests                                   */
/* -------------------------------------------------------------------------- */

describe('ReservationDetails', () => {
    describe('Data accuracy', () => {
        it('shows correct data', () => {
            const { getByText } = renderComponent();

            // Customer details preview
            expect(getByText('Nome').previousElementSibling).toHaveTextContent(
                __mockCustomers[0].name + ' ' + __mockCustomers[0].surname || '',
            );
            expect(getByText('Identificatore aggiuntivo').previousElementSibling).toHaveTextContent(
                __mockCustomers[0].additionalIdentifier || '---',
            );
            expect(getByText('Telefono').previousElementSibling).toHaveTextContent(__mockCustomers[0].phone || '---');
            expect(getByText('Note personali cliente').previousElementSibling).toHaveTextContent(
                __mockCustomers[0].notes || '---',
            );

            // Actual reservation details
            expect(getByText('Stato').previousElementSibling).toHaveTextContent('Completato');
            expect(getByText('Servizio').previousElementSibling).toHaveTextContent(__mockServices[0].name);
            expect(getByText('Dipendente assegnato').previousElementSibling).toHaveTextContent(__mockEmployees[0].name);
            expect(getByText('Inizio appuntamento').previousElementSibling).toHaveTextContent(
                __mockReservations[0].date.toLocaleDateString('it-IT', TIME_DATE_FORMAT),
            );
            expect(getByText('Fine appuntamento').previousElementSibling).toHaveTextContent(
                addMinutes(
                    new Date(__mockReservations[0].date.getTime()),
                    __mockServices[0].averageDuration,
                ).toLocaleDateString('it-IT', TIME_DATE_FORMAT),
            );
            expect(getByText('Note appuntamento').previousElementSibling).toHaveTextContent(
                __mockReservations[0].notes || '---',
            );
        });
    });

    describe('User experience', () => {
        it('prevents crashes if reservation not exists', () => {
            renderComponent({ match: { params: { reservationId: 'missing_id' } } });

            expect(spyTError).toBeCalledTimes(1);
        });

        it('prevents crashes if some information are missing', () => {
            // Change implementation of useStore in react-redux manual mock
            __setUseStore(() => ({
                getState: () => ({
                    database: {
                        customers: __mockCustomers,
                        employees: __mockEmployees,
                        services: [
                            /* es. services deleted */
                        ],
                        reservations: __mockReservations,
                    },
                }),
            }));

            renderComponent();

            expect(spyTError).toBeCalledTimes(1);

            // Restore useStore
            __setUseStore(() => ({
                getState: () => ({
                    database: {
                        customers: __mockCustomers,
                        employees: __mockEmployees,
                        services: __mockServices,
                        reservations: __mockReservations,
                    },
                }),
            }));
        });

        it('redirects users at "Go to reservation day" button click', () => {
            const { getByTestId } = renderComponent();

            fireEvent.click(getByTestId('go-to-day'));
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith(
                '/reservations?date=' + __mockReservations[0].date.toISOString().slice(0, 10),
            );
        });

        it('redirects users at "Customer details" button click', () => {
            const { getByTestId } = renderComponent();

            fireEvent.click(getByTestId('customer-details'));
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith('/customer-details/' + __mockCustomers[0].id);
        });

        it('redirects users at "Add Reservation" button click', () => {
            const { getByTestId } = renderComponent();
            fireEvent.click(getByTestId('add-reservation'));
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith('/add-reservation?customerId=' + __mockCustomers[0].id);
        });

        it('redirects users at "Edit" button click', () => {
            const { getByTestId } = renderComponent();

            fireEvent.click(getByTestId('edit-reservation'));
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith('/edit-reservation/' + __mockReservations[0].id);
        });

        it('display confirmation dialog at "Delete" button click', () => {
            const { getByTestId } = renderComponent();

            fireEvent.click(getByTestId('delete-reservation'));

            expect(mockOpenDialog).toBeCalledTimes(1);
        });
    });

    it('matches snapshot', () => {
        const snapshot = renderer
            .create(
                <StylesProvider generateClassName={createGenerateClassName({ disableGlobal: true })}>
                    {/* @ts-ignore */}
                    <ReservationDetails match={match} history={history} />
                </StylesProvider>,
            )
            .toJSON();
        expect(snapshot).toMatchSnapshot();
    });
});
