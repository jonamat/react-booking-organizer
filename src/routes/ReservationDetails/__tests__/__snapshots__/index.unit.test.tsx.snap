// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationDetails matches snapshot 1`] = `
<div
  className="jss1"
>
  <div
    className="jss5 jss6 jss25"
  >
    <div
      className="jss5 jss7 jss51 jss52"
    >
      <h6
        className="jss108 jss118"
      >
        Dettagli appuntamento
      </h6>
    </div>
    <div
      className="jss4"
    />
    <div
      className="jss5 jss7"
    >
      <button
        aria-describedby={null}
        className="jss158 jss149"
        data-testid="go-to-day"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Vai al giorno dell'appuntamento"
        type="button"
      >
        <span
          className="jss157"
        >
          <svg
            aria-hidden={true}
            className="jss161 jss162"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3h-1V2c0-.55-.45-1-1-1s-1 .45-1 1v1H8V2c0-.55-.45-1-1-1s-1 .45-1 1v1H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-1 16H6c-.55 0-1-.45-1-1V8h14v10c0 .55-.45 1-1 1zM8 10h3c.55 0 1 .45 1 1v3c0 .55-.45 1-1 1H8c-.55 0-1-.45-1-1v-3c0-.55.45-1 1-1z"
            />
          </svg>
        </span>
        <span
          className="jss193"
        />
      </button>
    </div>
    <div
      className="jss5 jss7"
    >
      <button
        aria-describedby={null}
        className="jss158 jss149"
        data-testid="add-reservation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Assegna nuovo appuntamento allo stesso cliente"
        type="button"
      >
        <span
          className="jss157"
        >
          <svg
            aria-hidden={true}
            className="jss161 jss162"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c-.55 0-1 .45-1 1v3H8c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </span>
        <span
          className="jss193"
        />
      </button>
    </div>
    <div
      className="jss5 jss7"
    >
      <button
        aria-describedby={null}
        className="jss158 jss149"
        data-testid="customer-details"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Vai alla scheda dettagli cliente"
        type="button"
      >
        <span
          className="jss157"
        >
          <svg
            aria-hidden={true}
            className="jss161 jss162"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3h-4.18C14.4 1.84 13.3 1 12 1s-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm1 14H8c-.55 0-1-.45-1-1s.45-1 1-1h5c.55 0 1 .45 1 1s-.45 1-1 1zm3-4H8c-.55 0-1-.45-1-1s.45-1 1-1h8c.55 0 1 .45 1 1s-.45 1-1 1zm0-4H8c-.55 0-1-.45-1-1s.45-1 1-1h8c.55 0 1 .45 1 1s-.45 1-1 1z"
            />
          </svg>
        </span>
        <span
          className="jss193"
        />
      </button>
    </div>
    <div
      className="jss5 jss7"
    >
      <button
        aria-describedby={null}
        className="jss158 jss149"
        data-testid="edit-reservation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Modifica appuntamento"
        type="button"
      >
        <span
          className="jss157"
        >
          <svg
            aria-hidden={true}
            className="jss161 jss162"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.46v3.04c0 .28.22.5.5.5h3.04c.13 0 .26-.05.35-.15L17.81 9.94l-3.75-3.75L3.15 17.1c-.1.1-.15.22-.15.36zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        <span
          className="jss193"
        />
      </button>
    </div>
    <div
      className="jss5 jss7"
    >
      <button
        aria-describedby={null}
        className="jss158 jss149"
        data-testid="delete-reservation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Elimina appuntamento"
        type="button"
      >
        <span
          className="jss157"
        >
          <svg
            aria-hidden={true}
            className="jss161 jss162"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v10zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
            />
          </svg>
        </span>
        <span
          className="jss193"
        />
      </button>
    </div>
  </div>
  <div
    className="jss5 jss6"
  >
    <div
      className="jss5 jss7 jss51"
    >
      <p
        className="jss108 jss3 jss110"
      >
        Cliente
      </p>
    </div>
    <div
      className="jss5 jss7 jss51 jss59"
    >
      <ul
        className="jss170"
      >
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3h-4.18C14.4 1.84 13.3 1 12 1s-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm0 4c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1.4c0-2 4-3.1 6-3.1s6 1.1 6 3.1V19z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              Customer name Surname
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Nome
            </p>
          </div>
        </li>
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 11c-1.33 0-4 .67-4 2v.16c.97 1.12 2.4 1.84 4 1.84s3.03-.72 4-1.84V13c0-1.33-2.67-2-4-2zm0-1c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0-8c4.2 0 8 3.22 8 8.2 0 3.18-2.45 6.92-7.34 11.23-.38.33-.95.33-1.33 0C6.45 17.12 4 13.38 4 10.2 4 5.22 7.8 2 12 2z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              ---
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Identificatore aggiuntivo
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="jss5 jss7 jss51 jss59"
    >
      <ul
        className="jss170"
      >
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              ---
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Telefono
            </p>
          </div>
        </li>
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 12c-.55 0-1 .45-1 1v5.22c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1h5c.55 0 1-.45 1-1s-.45-1-1-1H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-6c0-.55-.45-1-1-1zM21.02 5H19V2.98c0-.54-.44-.98-.98-.98h-.03c-.55 0-.99.44-.99.98V5h-2.01c-.54 0-.98.44-.99.98v.03c0 .55.44.99.99.99H17v2.01c0 .54.44.99.99.98h.03c.54 0 .98-.44.98-.98V7h2.02c.54 0 .98-.44.98-.98v-.04c0-.54-.44-.98-.98-.98z"
              />
              <path
                d="M14 9H8c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1zM14 12H8c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1zM14 15H8c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              ---
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Note personali cliente
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="jss5 jss7 jss51"
    >
      <p
        className="jss108 jss3 jss110"
      >
        Appuntamento
      </p>
    </div>
    <div
      className="jss5 jss7 jss51 jss59"
    >
      <ul
        className="jss170"
      >
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm-.22-13h-.06c-.4 0-.72.32-.72.72v4.72c0 .35.18.68.49.86l4.15 2.49c.34.2.78.1.98-.24.21-.34.1-.79-.25-.99l-3.87-2.3V7.72c0-.4-.32-.72-.72-.72z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              Completato
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Stato
            </p>
          </div>
        </li>
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.09 2.91C10.08.9 7.07.49 4.65 1.67L8.28 5.3c.39.39.39 1.02 0 1.41L6.69 8.3c-.39.4-1.02.4-1.41 0L1.65 4.67C.48 7.1.89 10.09 2.9 12.1c1.86 1.86 4.58 2.35 6.89 1.48l7.96 7.96c1.03 1.03 2.69 1.03 3.71 0 1.03-1.03 1.03-2.69 0-3.71L13.54 9.9c.92-2.34.44-5.1-1.45-6.99z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              fake service 1
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Servizio
            </p>
          </div>
        </li>
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 11.75c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zm6 0c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8 0-.29.02-.58.05-.86 2.36-1.05 4.23-2.98 5.21-5.37C11.07 8.33 14.05 10 17.42 10c.78 0 1.53-.09 2.25-.26.21.71.33 1.47.33 2.26 0 4.41-3.59 8-8 8z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              Employee name 1
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Dipendente assegnato
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="jss5 jss7 jss51 jss59"
    >
      <ul
        className="jss170"
      >
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 1h-4c-.55 0-1 .45-1 1s.45 1 1 1h4c.55 0 1-.45 1-1s-.45-1-1-1zm-2 13c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .55.45 1 1 1zm7.03-6.61l.75-.75c.38-.38.39-1.01 0-1.4l-.01-.01c-.39-.39-1.01-.38-1.4 0l-.75.75C16.07 4.74 14.12 4 12 4c-4.8 0-8.88 3.96-9 8.76C2.87 17.84 6.94 22 12 22c4.98 0 9-4.03 9-9 0-2.12-.74-4.07-1.97-5.61zM12 20c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              sabato 1 gennaio 2000, 10:00
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Inizio appuntamento
            </p>
          </div>
        </li>
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 3h4c.55 0 1-.45 1-1s-.45-1-1-1h-4c-.55 0-1 .45-1 1s.45 1 1 1zm2 5c-.51 0-.92.39-.98.89L13 10.86V9c0-.55-.45-1-1-1zm0-2c3.87 0 7 3.13 7 7 0 1.12-.27 2.18-.74 3.12l1.47 1.47C20.53 16.25 21 14.68 21 13c0-2.12-.74-4.07-1.97-5.61l.75-.75c.38-.38.39-1.01 0-1.4l-.01-.01c-.39-.39-1.01-.38-1.4 0l-.75.75C16.07 4.74 14.12 4 12 4c-1.66 0-3.22.48-4.57 1.29l1.45 1.45c.94-.47 2-.74 3.12-.74zm8.19 14.88L3.87 4.56a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41l2.07 2.07C3.61 9.42 3.05 11.03 3 12.76 2.87 17.84 6.94 22 12 22c1.84 0 3.55-.55 4.98-1.5l1.79 1.79c.39.39 1.02.39 1.41 0 .4-.39.4-1.02.01-1.41zM12 20c-3.87 0-7-3.13-7-7 0-1.29.35-2.49.96-3.52l9.57 9.57c-1.04.6-2.24.95-3.53.95z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              sabato 1 gennaio 2000, 10:30
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Fine appuntamento
            </p>
          </div>
        </li>
        <li
          className="jss174 jss181"
          disabled={false}
        >
          <div
            className="jss185"
          >
            <svg
              aria-hidden={true}
              className="jss161 jss162"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 12c-.55 0-1 .45-1 1v5.22c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1h5c.55 0 1-.45 1-1s-.45-1-1-1H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-6c0-.55-.45-1-1-1zM21.02 5H19V2.98c0-.54-.44-.98-.98-.98h-.03c-.55 0-.99.44-.99.98V5h-2.01c-.54 0-.98.44-.99.98v.03c0 .55.44.99.99.99H17v2.01c0 .54.44.99.99.98h.03c.54 0 .98-.44.98-.98V7h2.02c.54 0 .98-.44.98-.98v-.04c0-.54-.44-.98-.98-.98z"
              />
              <path
                d="M14 9H8c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1zM14 12H8c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1zM14 15H8c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1z"
              />
            </svg>
          </div>
          <div
            className="jss187 jss188"
          >
            <span
              className="jss108 jss191 jss110 jss137"
            >
              something
            </span>
            <p
              className="jss108 jss192 jss109 jss134 jss137"
            >
              Note appuntamento
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
